drivers/led/tlc59108.c

```diff
@@ -0,0 +1,245 @@
+	struct led_data *dev_data = &data->dev_data;
+	uint8_t gdc, gfrq;
+	uint32_t period;
+
+	period = delay_on + delay_off;
+
+	if (led > TLC59108_MAX_LED) {
+		return -EINVAL;
+	}
+
+	if (period < dev_data->min_period || period > dev_data->max_period) {
+		return -EINVAL;
+	}
+
+	/*
+	 * From manual:
+	 * duty cycle = (GDC / 256) ->
+	 *	(time_on / period) = (GDC / 256) ->
+	 *		GDC = ((time_on * 256) / period)
+	 */
+	gdc = delay_on * 256U / period;
+	if (i2c_reg_write_byte(config->i2c_dev, config->i2c_addr,
+				TLC59108_GRPPWM,
+				gdc)) {
+		LOG_ERR("LED reg 0x%x write failed", TLC59108_GRPPWM);
+		return -EIO;
+	}
+
+	/*
+	 * From manual:
+	 * period = ((GFRQ + 1) / 24) in seconds.
+	 * So, period (in ms) = (((GFRQ + 1) / 24) * 1000) ->
+	 *		GFRQ = ((period * 24 / 1000) - 1)
+	 */
+	gfrq = (period * 24U / 1000) - 1;
+	if (i2c_reg_write_byte(config->i2c_dev, config->i2c_addr,
+				TLC59108_GRPFREQ,
+				gfrq)) {
+		LOG_ERR("LED reg 0x%x write failed", TLC59108_GRPFREQ);
+		return -EIO;
+	}
+
+	/* Enable blinking mode */
+	if (i2c_reg_update_byte(config->i2c_dev, config->i2c_addr,
+				TLC59108_MODE2,
+				TLC59108_MODE2_DMBLNK,
+				TLC59108_MODE2_DMBLNK)) {
+		LOG_ERR("LED reg 0x%x update failed", TLC59108_MODE2);
+		return -EIO;
+	}
+
+	/* Select the GRPPWM source to drive the LED outpout */
+	return tlc59108_set_ledout(dev, led, TLC59108_LED_GRP_PWM);
+}
+
+static int tlc59108_led_set_brightness(const struct device *dev, uint32_t led,
+		uint8_t value)
+{
+	const struct tlc59108_cfg *config = dev->config;
+	struct tlc59108_data *data = dev->data;
+	struct led_data *dev_data = &data->dev_data;
+	uint8_t val;
+
+	if (led > TLC59108_MAX_LED) {
+		return -EINVAL;
+	}
+
+	if (value < dev_data->min_brightness ||
+	    value > dev_data->max_brightness) {
+		return -EINVAL;
+	}
+
+	/* Set the LED brightness value */
+	val = (value * 255U) / dev_data->max_brightness;
+	if (i2c_reg_write_byte(config->i2c_dev, config->i2c_addr,
+				TLC59108_PWM_BASE + led,
+				val)) {
+		LOG_ERR("LED 0x%x reg write failed", TLC59108_PWM_BASE + led);
+		return -EIO;
+	}
+
+	/* Set the LED driver to be controlled through its PWMx register. */
+	return tlc59108_set_ledout(dev, led, TLC59108_LED_PWM);
+}
+
+static inline int tlc59108_led_on(const struct device *dev, uint32_t led)
+{
+	if (led > TLC59108_MAX_LED) {
+		return -EINVAL;
+	}
+
+	/* Set LED state to ON */
+	return tlc59108_set_ledout(dev, led, TLC59108_LED_ON);
+}
+
+static inline int tlc59108_led_off(const struct device *dev, uint32_t led)
+{
+	if (led > TLC59108_MAX_LED) {
+		return -EINVAL;
+	}
+
+	/* Set LED state to OFF */
+	return tlc59108_set_ledout(dev, led, TLC59108_LED_OFF);
+}
+
+static int tlc59108_led_init(const struct device *dev)
+{
+	const struct tlc59108_cfg *config = dev->config;
+	struct tlc59108_data *data = dev->data;
+	struct led_data *dev_data = &data->dev_data;
+
+	if (!device_is_ready(config->i2c_dev)) {
+		LOG_ERR("I2C bus device not ready");
```

Comment:
```suggestion
		LOG_ERR("I2C bus device %s is not ready", config->i2c_dev->name);
```

Including device names in error logs related to devices eases debugging.